passes/src/Luna/Pass/Inference/ErrorPropagation.hs

```diff
@@ -27,13 +27,21 @@ run = do
     mapM_ propagateErrors roots
 
 propagateErrors :: (MonadRef m, MonadPassManager m) => Expr Draft -> SubPass ErrorPropagation m ()
-propagateErrors expr = matchExpr expr $ \case
-    Seq a b   -> do
-        propagateErrors =<< source a
-        propagateErrors =<< source b
-    _ -> do
-        inpErrors <- fmap concat $ mapM (getErrors <=< source) =<< inputs expr
-        modifyLayer_ @Errors expr $ nub . (++ inpErrors)
+propagateErrors expr = do
+    matchExpr expr $ \case
+        Seq a b   -> do
+            propagateErrors =<< source a
+            propagateErrors =<< source b
+        -- FIXME[MM]: this line causes correct reporting of compile errors
+        --            we don't want that until interpreter can cope with them
+        --            and interpret the rest of the program
+        -- ASGFunction _ _ g -> propagateErrors =<< source g
```

Comment:
The errors are need both in `Seq`s and in the `ASGFunction`. The latter for properly handling them when importing a function.